awslogs/core.py

```diff
@@ -248,13 +248,16 @@ def parse_datetime(self, datetime_text):
         if not datetime_text:
             return None
 
-        ago_regexp = r'(\d+)\s?(m|minute|minutes|h|hour|hours|d|day|days|w|weeks|weeks)(?: ago)?'
+        ago_regexp = r'([+-]?([0-9]*[.])?[0-9]+)\s?(s|sec|secs|second|seconds|m|minute|minutes|h|hour|hours|d|day|days|w|weeks|weeks)(?: ago)?'
```

Comment:
I think I agree that the +/- is redundant.

The digit pattern may be a bit more efficient (not a practical concern) as `([0-9]+(?:\.[0-9]+)?)` - it would also avoid the additional matching group, so you just have (amount, unit) again.

To be consistent the additional time spans would be `s|second|seconds`, as there's nothing like `hr|hrs`.

The line is getting long enough to think of changing to `s|seconds?|m|minutes?|…`.

_p.s. it looks like there was a bug previously, was the pattern ends with `weeks|weeks` instead of `week|weeks`. It would also be good to add `$` to the end of the pattern._

Reply:
- I agree the +/- is a bit too much.
- I do agree with @Code0x58 about the regexp and ``s|second|seconds``